zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -673,6 +716,10 @@ impl UpdateWith<Option<transaction::JoinSplitData<Groth16Proof>>> for Chain {
         joinsplit_data: &Option<transaction::JoinSplitData<Groth16Proof>>,
     ) -> Result<(), ValidateContextError> {
         if let Some(joinsplit_data) = joinsplit_data {
+            for cm_u in joinsplit_data.note_commitments() {
+                self.sprout_note_commitment_tree.append(*cm_u)?;
+            }
```

Comment:
The `_u` here indicates that the Sapling note commitment is specified only by the u-coordinate of the Jubjub curve point (u, v). Since Sprout note commitments are just SHA-256 hash values, we can remove the `_u`:

```suggestion
            for cm in joinsplit_data.note_commitments() {
                self.sprout_note_commitment_tree.append(*cm)?;
            }
```

Reply:
I was wondering about the `_u`. I think the spec uses it as well, now I know.

Reply:
The spec only uses it for Sapling. For Orchard there's a similar thing with cm<sub>x</sub>, which is only the short Weierstrass x-coordinate.